setup.py

```diff
@@ -0,0 +1,28 @@
+#! /usr/bin/env python3
+
+from setuptools import setup, find_packages
+
+setup(
+    name="canonicalwebteam.discourse",
+    version="0.1",
+    author="Canonical webteam",
+    author_email="webteam@canonical.com",
+    url="https://github.com/canonical-webteam/canonicalwebteam.docs",
+    description=(
+        "Flask extension to integrate discourse content generated "
+        "to docs to your website."
+    ),
+    packages=find_packages(),
+    long_description=open("README.md").read(),
+    install_requires=[
```

Comment:
Module dependencies should be as loose as possible, because client applications can ultimately only install one version of a dependency, and so if two dependencies require different specific versions of of the same module then it has no way to satisfy both (so it will just do it's best, but might do the wrong thing).

From https://packaging.python.org/discussions/install-requires-vs-requirements/:

> It is not considered best practice to use install_requires to pin dependencies to specific versions, or to specify sub-dependencies (i.e. dependencies of your dependencies). This is overly-restrictive, and prevents the user from gaining the benefit of dependency upgrades.

Now obviously this attitude carries risk - the only way to be sure our version ranges are all supported is to test not only with every dependency version in the range, but also every possible combination of dependencies - a possibly incredibly large number of permutations. I think [talisker](https://github.com/canonical-ols/talisker/) sort of tries to do this, but it's a lot of work.

Given that this is currently mostly a library for our own use, and that we have limited time to work on it, I suggest we instead take the opposite attitude and not restrict versions until we know them to be broken, e.g.:

``` python3
install_requires=["flask", "canonicalwebteam.http", ...]
```

Then we can discover incompatible versions as we run our projects, and decide at that point whether we want to fix the library to support that version or restrict the versions in this file.

Of course if we *know* right away that certain versions won't work, we should restrict it right away - e.g. we might know that we need `"flask>=1"` (I don't know if that's true).

Reply:
done